logr.go

```diff
@@ -169,22 +165,62 @@ type Logger interface {
 	// triggered this log line, if present.
 	Error(err error, msg string, keysAndValues ...interface{})
 
-	// V returns an Logger value for a specific verbosity level, relative to
-	// this Logger.  In other words, V values are additive.  V higher verbosity
-	// level means a log message is less important.  It's illegal to pass a log
-	// level less than zero.
-	V(level int) Logger
-
 	// WithValues adds some key-value pairs of context to a logger.
 	// See Info for documentation on how key/value pairs work.
-	WithValues(keysAndValues ...interface{}) Logger
+	WithValues(keysAndValues ...interface{}) LogSink
 
 	// WithName adds a new element to the logger's name.
 	// Successive calls with WithName continue to append
 	// suffixes to the logger's name.  It's strongly recommended
 	// that name segments contain only letters, digits, and hyphens
 	// (see the package documentation for more information).
-	WithName(name string) Logger
+	WithName(name string) LogSink
+}
+
+func New(level int, sink LogSink) Logger {
```

Comment:
It would be nicer to make the signature `New(sink LogSink) Logger`. The caller can always do this:

```go
logger := New(someSink).V(1)
```

Reply:
Good point.  Early drafts needed this, but not any more.